drivers/gpio/gpio_emul.c

```diff
@@ -406,41 +418,38 @@ static int gpio_emul_pin_configure(const struct device *port, gpio_pin_t pin,
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&drv_data->mu, K_FOREVER);
+	key = k_spin_lock(&drv_data->lock);
```

Comment:
Can you do a second pass and adjust the API doc to use `lock` instead of `mu`?

Reply:
You men the internal documentation of the data struct?

Reply:
Yeah, just for consistence